### PR TITLE
Cleanup service creation output

### PR DESCRIPTION
**Pull request by cdrage**
**What kind of PR is this?**

/kind cleanup
/kind enhancement

**What does does this PR do / why we need it**:

Cleans up the output for service creation and adds an output of what to
do next:

```sh
Deploying service mongodb of type: dh-mongodb-apb
 ✓  Deploying service [78ms]
 ✓  Service 'mongodb' was created and is currently provisioning
View the status of the provision by executing 'odo service list'
Optionally, link dh-mongodb-apb to your component by running: 'odo link <component-name>'
```

**Which issue(s) this PR fixes**:

N/A

**How to test changes / Special notes to the reviewer**:

```sh
make bin && ./odo service create dh-mongodb-apb mongodb --plan persistent -p DATABASE_SERVICE_NAME=mongodb -p MEMORY_LIMIT=512Mi -p MONGODB_DATABASE=sampledb -p VOLUME_CAPACITY=1Gi
```

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: pkg/odo/cli/service/create.go [hunk 1]
```diff
@@ -240,10 +242,13 @@ func (o *ServiceCreateOptions) Run() (err error) {
 			log.Successf(`Service '%s' is ready for use`, o.ServiceName)
 		}
 	} else {
-		log.Successf(`Service '%s' was created`, o.ServiceName)
-		log.Info(`Progress of the provisioning will not be reported and might take a long time.
-You can see the current status by executing 'odo service list'`)
+		log.Successf(`Service '%s' was created and is currently provisioning`, o.ServiceName)
```
Comment (girishramnani) [inline pkg/odo/cli/service/create.go]: is currently being provisioned?

Comment (mik-dass) [reply pkg/odo/cli/service/create.go]: > is currently being provisioned?

Sounds better to me :+1:

**Comment by dharmit** [comment]
There's no issue linked to this PR. Do we have any feedback about current output being unclear/unhelpful? Was there a discussion about making changes to the output of this command? 

I'm just trying to understand if the changes proposed by this PR have been discussed. Otherwise, a few weeks later, someone might come up with yet another PR that's proposing a change to the same output.

cc @girishramnani @kadel

**Comment by cdrage** [comment]
> There's no issue linked to this PR. Do we have any feedback about current output being unclear/unhelpful? Was there a discussion about making changes to the output of this command?
> 
> I'm just trying to understand if the changes proposed by this PR have been discussed. Otherwise, a few weeks later, someone might come up with yet another PR that's proposing a change to the same output.
> 
> cc @girishramnani @kadel

I noticed that there was no aforementioned 'next steps' for the service creation output. However, @kadel and @girishramnani you can have a look at this and give your input! :)

**Comment by dharmit** [comment]
> I noticed that there was no aforementioned 'next steps' for the service creation output. 

:+1: 

/approve

**Comment by dharmit** [comment]
```sh

[91m[1mSummarizing 1 Failure:[0m

[91m[1m[Fail] [0m[90modo project command tests [0m[0mwhen running project command app parameter in directory that doesn't contain .odo config directory [0m[91m[1m[It] should successfully execute list along with machine readable output [0m
[37m/go/src/github.com/openshift/odo/tests/integration/project/cmd_project_test.go:44[0m

[1m[91mRan 2 of 2 Specs in 14.102 seconds[0m
[1m[91mFAIL![0m -- [32m[1m1 Passed[0m | [91m[1m1 Failed[0m | [33m[1m0 Pending[0m | [36m[1m0 Skipped[0m
--- FAIL: TestProject (14.10s)
FAIL
```
#2298 

/test v4.1-integration

**Comment by dharmit** [comment]
Gateway Time-out errors all over the place

```
error: unable to extract layer sha256:034013f8264937c7028fac94f64410b2d0d7d0b8b2d79d8919a06391bb69a485 from registry.svc.ci.openshift.org/ci-op-vjz9ftzv/release:latest: received unexpected HTTP status: 504 Gateway Time-out
```

```
error: unable to read image registry.svc.ci.openshift.org/ci-op-61lbk9in/stable@sha256:cf46556b3ab824a9a3b057a40400ee9f4adf6ae4b43a66183e8e651fc303cb29: received unexpected HTTP status: 504 Gateway Time-out
```

```
error: unable to read image registry.svc.ci.openshift.org/ci-op-61lbk9in/stable@sha256:cf46556b3ab824a9a3b057a40400ee9f4adf6ae4b43a66183e8e651fc303cb29: received unexpected HTTP status: 504 Gateway Time-out
```

/retest

**Comment by dharmit** [comment]
@amitkrout I don't think this is a flake. Can you please check:
```
Summarizing 1 Failure:

[Fail] odo core beta flow when --context flag is used [It] create local java component and push code 
/home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_run.go:33

Ran 6 of 15 Specs in 97.461 seconds
FAIL! -- 5 Passed | 1 Failed | 0 Pending | 9 Skipped
```

File: pkg/odo/cli/service/create.go (hunk 1 above)
Comment (girishramnani) [reply pkg/odo/cli/service/create.go]: @cdrage please address this comment

**Comment by amitkrout** [comment]
> @amitkrout I don't think this is a flake. Can you please check:
> 
> ```
> Summarizing 1 Failure:
> 
> [Fail] odo core beta flow when --context flag is used [It] create local java component and push code 
> /home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_run.go:33
> 
> Ran 6 of 15 Specs in 97.461 seconds
> FAIL! -- 5 Passed | 1 Failed | 0 Pending | 9 Skipped
> ```

May be related to #2088. 
```
Running oc with args [oc get is -n openshift -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}']
[oc] error: Missing or incomplete configuration info.  Please login or point to an existing, complete config file:
[oc] 
[oc]   1. Via the command-line flag --config
[oc]   2. Via the KUBECONFIG environment variable
[oc]   3. In your home directory as ~/.kube/config
[oc] 
[oc] To view or setup config directly use the 'config' command.
```

File: pkg/odo/cli/service/create.go (hunk 1 above)
Comment (cdrage) [reply pkg/odo/cli/service/create.go]: Thanks! I've fixed this.

**Comment by cdrage** [comment]
@dharmit @amitkrout @girishramnani Updated with the last message / output change. Thanks!

**Pull request merged**